webapp/graphite/errors.py

```diff
@@ -16,6 +81,8 @@ def new_f(*args, **kwargs):
         try:
             return f(*args, **kwargs)
         except InputParameterError as e:
-            return HttpResponseBadRequest('Bad Request: {err}'.format(err=e))
+            msg = e.describe()
+            log.warning(msg)
```

Comment:
This should probably be `log.warning("%s", msg)`.

The reason is that any standard-like logging function which takes args for formatting, treats its first argument as a format string. If that string happens to unexpectedly contain `%` it breaks. So you generally want the first argument to always be a string literal, or one of few obvious possibilities, to be sure that logging statement is robust.

Reply:
thanks, fixed